Zenodo.js

```diff
@@ -120,36 +127,35 @@ function doWeb(doc, url) {
 function scrape(doc, url) {
 	var abstract = ZU.xpathText(doc, '//meta[@name="description"]/@content');
 	var doi = ZU.xpathText(doc, '//meta[@name="citation_doi"]/@content');
-	var schemaType = ZU.xpathText(doc, '//div[contains(@class, "record-detail")]//div[@itemscope]/@itemtype');
 	var pdfURL = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
 	var tags = ZU.xpath(doc, '//meta[@name="citation_keywords"]');
 	var cslURL = url.replace(/#.+/, "").replace(/\?.+/, "").replace(/\/export\/.+/, "") + "/export/csl";
-	//Z.debug(cslURL)
-	//Z.debug(schemaType)
+	// Z.debug(cslURL)
 	// use CSL JSON translator
-	ZU.processDocuments(cslURL, function(newDoc){
+	ZU.processDocuments(cslURL, function (newDoc) {
 		var text = ZU.xpathText(newDoc, '//h3/following-sibling::pre');
-		//Z.debug(text)
+		// Z. debug(text);
 		text = text.replace(/publisher_place/, "publisher-place");
 		text = text.replace(/container_title/, "container-title");
 
 		var trans = Zotero.loadTranslator('import');
 		// CSL JSON
 		trans.setTranslator('bc03b4fe-436d-4a1f-ba59-de4d2d7a63f7');
 		trans.setString(text);
-		trans.setHandler("itemDone", function(obj, item) {
+		trans.setHandler("itemDone", function (obj, item) {
+			item.itemType = detectWeb(doc, url);
 			// The "note" field of CSL maps to Extra. Put it in a note instead
 			if (item.extra) {
-				item.notes.push({"note": item.extra});
+				item.notes.push({ note: item.extra });
 				item.extra = "";
 			}
-
-			if (!item.DOI && doi) {
+			if (!ZU.fieldIsValidForType('DOI', item.itemType) && doi) {
 				item.extra = "DOI: " + doi;
 			}
-			//workaround while we don't have proper item type for data
-			//if (schemaType && schemaType.includes("Dataset")) {
-			if (ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
+
+			// workaround for pre 6.0.24 versions that don't have proper item type for data
+			// if (schemaType && schemaType.includes("Dataset")) {
+			if (!datasetType == "dataset" && ZU.xpathText(doc, '//span[@class="pull-right"]/span[contains(@class, "label-default") and contains(., "Dataset")]')) {
```

Comment:
> !datasetType == "dataset"

This looks like a mistake.

Reply:
I don't think it is? `datasetType` is [defined here](https://github.com/zotero/translators/pull/3021/files#diff-23ef15323eb9526928ac756f18c84013122c75bdb657077836608330105c9a77R38) and this code should only run where dataset isn't a valid item type, which this tests for.

Reply:
But you're testing whether `!dataset` is equal to a string, which will always be false.